staging/vhost-device-gpu/src/virtio_gpu.rs

```diff
@@ -571,12 +574,7 @@ impl VirtioGpu for RutabagaVirtioGpu {
         self.rutabaga
             .resource_create_3d(resource_id, resource_create_3d)?;
 
-        let resource = VirtioGpuResource::new(
-            resource_id,
-            resource_create_3d.width,
-            resource_create_3d.height,
-            0,
-        );
+        let resource = VirtioGpuResource::new(resource_create_3d.width, resource_create_3d.height);
```

Comment:
I think it will be better to use the resource_id here so that it can be used when flushing to the display in `fn flush_resource()`. So instead of parsing `resource_id: u32,` use `resource: &mut VirtioGpuResource,` and parse it as an argument `resource.resource_id` to `self.read_2d_resource()`

Reply:
I am not sure if I understand what you mean. We already have the `resource_id` in `flush_resource`. If you mean we should pass `&mut VirtioGpuResource` to `read_2d_resource` then we could do that*. But this will still require passing the `rect` as a seperate argument, because we might want to transfer only a portion of the resource when using virglrenderer. And now we will have `if self.renderer == GpuMode::ModeGfxstream` check in 2 places - the first place is where we create the buffer to pass as `output` to `read_2d_resource` and the second one is inside `read_2d_resource`.

Another option is to just give up and always read the whole resource using the `resource.width`,  `resource.height` even with `virglrenderer`. This is probably less efficient, but eventually we will hopefully be using `export_blob()` with qemu dmabuf scanout anyway, which won't require `transfer_read` at all. 

*: this would result in borrow checker error, since we would have both `&mut self` and  `&mut VirtioGpuResource` and the `&mut VirtioGpuResource` is a reference into `self`. This could be simply solved though by copying `VirtioGpuResource`, which isn't a problem because the struct doesn't contain anything expensive to copy just some integers.

Reply:
> If you mean we should pass `&mut VirtioGpuResource` to `read_2d_resource` then we could do that*

this is what i meant.

> Another option is to just give up and always read the whole resource using the `resource.width`, `resource.height` even with `virglrenderer`. This is probably less efficient, but eventually we will hopefully be using `export_blob()` with qemu dmabuf scanout anyway, which won't require `transfer_read` at all.

atleast this is what qemu does for the inbuilt virtio-gpu. I agree we can do this also considering the fact that our main focus is gfxstream support.




> this would result in borrow checker error, since we would have both `&mut self` and `&mut VirtioGpuResource` and the `&mut VirtioGpuResource` is a reference into `self`. This could be simply solved though by copying `VirtioGpuResource`, which isn't a problem because the struct doesn't contain anything expensive to copy just some integers.

I would prefer this because it offers some consistency in the use of resource_id and we are not mistakenly using 2 different resource_ids

Reply:
> I would prefer this because it offers some consistency in the use of resource_id and we are not mistakenly using 2 different resource_ids
I don't see how this reduces the chance of mixing up 2 different resource_ids, I don't think there are many places where we have more than 1 in scope.

I'll make the change like you are suggesting, the change might be beneficial for blob resource scanout, though I am not sure yet.